src/Embeddings/VectorStores/Doctrine/DoctrineVectorStore.php

```diff
@@ -68,8 +73,8 @@ public function similaritySearch(array $embedding, int $k = 4, array $additional
         $repository = $this->entityManager->getRepository($this->entityClassName);
         $qb = $repository
             ->createQueryBuilder('e')
-            ->orderBy('L2_DISTANCE(e.pgembedding, :embeddingString)', 'ASC')
-            ->setParameter('embeddingString', $this->formatEmbeddingForPostgresql($embedding))
+            ->orderBy('L2_DISTANCE(e.embedding, :embeddingString)', 'ASC')
+            ->setParameter('embeddingString', $embedding, VectorType::VECTOR)
```

Comment:
it's not a string anymore

```suggestion
            ->orderBy('L2_DISTANCE(e.embedding, :vector)', 'ASC')
            ->setParameter('vector', $embedding, VectorType::VECTOR)
```